sync.py

```diff
@@ -199,12 +203,31 @@ def sync_metarepo(clone_dir, name, submodules):
         logging.info('Meta repository %s requires no changes', name)
 
 
-def handle_event(args, payload):
+def handle_events(args, event_queue):
+    # wait for an event
+    event_queue.get()
+    logging.info('Starting meta repository update')
+
+    # discard any other piled up events
+    while not event_queue.empty():
+        try:
+            event_queue.get(blocking=False)
+        except queue.Empty:
+            continue
+        event_queue.task_done()
+
+    # update the meta repos
     repos = list_repos(args.token)
     repos_by_topic = group_repos_by_topic(repos)
-    for name, topics in REPOS.items():
-        submodules = repos_for_topics(repos_by_topic, topics)
-        sync_metarepo(args.dir, name, submodules)
+    with concurrent.futures.ThreadPoolExecutor() as pool:
```

Comment:
Isn't it possible that `handle_event` is called again before the previous call's sync pool has finished? (Especially as the default interval is 3 seconds, and git pushes usually take that sort of time.) While that's not really a problem, I think it makes the structure more complicated than if we just synced the submodules in a blocking way.

Reply:
No, there's an implicit join at the end of the `with` block:

https://docs.python.org/dev/library/concurrent.futures.html#concurrent.futures.Executor.shutdown